VisualRust/RustCommentSelectionCommandHandler.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.TextManager.Interop;
+
+namespace VisualRust
+{
+    class RustCommentSelectionCommandHandler : VSCommandTarget<VSConstants.VSStd2KCmdID>
+    {
+        readonly ITextBuffer Buffer;
+        public RustCommentSelectionCommandHandler(IVsTextView vsTextView, IWpfTextView textView)
+            : base(vsTextView, textView)
+        {
+            Buffer = textView.TextBuffer;
+        }
+
+        protected override bool Execute(VSConstants.VSStd2KCmdID command, uint options, IntPtr pvaIn, IntPtr pvaOut)
+        {
+            var snapshot = Buffer.CurrentSnapshot;
+            int start = TextView.Selection.Start.Position.Position;
+            int end = TextView.Selection.End.Position.Position;
+
+            // this is what we will store start offset in to get maximal indentation amount
+            int? insertStartOffset = null;
+
+            using (var edit = Buffer.CreateEdit())
+            {
+                while (start <= end)
```

Comment:
When `start == end == <end of line>`, this will loop forever (you can see this eg. when your document has a single character and your cursor is behind it). Same with `GetOffset(...)`.